paimon-core/src/main/java/org/apache/paimon/encryption/KeyMetadata.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.encryption;
+
+import org.apache.paimon.types.DataField;
+import org.apache.paimon.types.RowType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static org.apache.paimon.utils.SerializationUtils.newBytesType;
+import static org.apache.paimon.utils.SerializationUtils.newStringType;
+
+/** Wrap information about the key. */
+public class KeyMetadata {
+
+    private final String keyId;
+    private final byte[] encryptedKey;
+    private final byte[] plaintextKey;
```

Comment:
Why we need `plaintextKey`?

Reply:
When using the file writer to write data, the plaintext data key is required for encryption. `EncryptionManager#createLocalDataKey`  method is used to get a wrapper for the data key ,`KeyMetadata` will be passed as a parameter between methods 

`KeyMetadata` will be persisted, but the plaintext data key cannot be stored.